xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -337,7 +337,17 @@ void CGUIDialogAddonInfo::OnInstall()
   if (!m_item->HasAddonInfo())
     return;
 
-  if (m_localAddon)
+  const auto& itemAddonInfo = m_item->GetAddonInfo();
+
+  const std::string& addonId = itemAddonInfo->ID();
+  const std::string& origin = itemAddonInfo->Origin();
+  const AddonVersion& version = itemAddonInfo->Version();
```

Comment:
Why have you moved `addonId` and `version`? You only need `origin` here...

Reply:
just wanted to get this fixed quickly and moved the whole block.
but sure both suggestions legit and coming in soon, thanks